models/_sector/dex/trades/optimism/platforms/wombat_exchange_optimism_base_trades.sql

```diff
@@ -0,0 +1,67 @@
+{{ config(
+    
+    schema = 'wombat_exchange_optimism',
+    alias = 'base_trades',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index']
+    )
+}}
+  
+
+
+{% set project_start_date = '2023-09-22' %}
+{% set wombat_exchange_optimism_evt_trade_tables = [
+    source('wombat_exchange_optimism', 'CrossChainPool_evt_SwapV2')
+] %}
+
+WITH dexs AS
+(
+    {% for evt_trade_table in wombat_exchange_optimism_evt_trade_tables %}
+
+    SELECT
+        t.evt_block_number AS block_number
+        , t.evt_block_time AS block_time
+        , t.evt_tx_to AS taker
+        , t.evt_tx_from AS maker
+        , t.toAmount AS token_sold_amount_raw
+        , t.fromAmount AS token_bought_amount_raw
+        , t.toToken AS token_sold_address
+        , t.fromToken AS token_bought_address
+        , t.contract_address AS project_contract_address
+        , t.evt_tx_hash AS tx_hash
+        , t.evt_index AS evt_index
```

Comment:
We may use the `generic_spot_v2_compatible_trades` macro to implement `wombat_exchange_optimism.base_trades`.
Do you want to have a try?
https://github.com/duneanalytics/spellbook/blob/main/macros/models/_sector/dex/generic_spot_compatible_trades.sql#L52

You can use this as an example:
https://github.com/duneanalytics/spellbook/blob/main/models/_sector/dex/trades/bnb/platforms/wombat_bnb_base_trades.sql

Also please recheck if the bought/sold tokens are flipped.